### PR TITLE
Add basic classes

### PR DESCRIPTION
**Pull request by chrisg564**
- Block, switch, and track class 
- More to be implemented but I think this is a good way of sharing information about the track between modules
- Also added json file for each track line (still need to format infrastructure lines)

### CODE REVIEW

**Comment by chrisg564** [comment]
@cag185 I think it would be smartest for us both to work off the same classes and obviously expand them to suit our needs. This way we are both in sync with the structure of the blocks.

What are your thoughts on approaching the classes like this?

**Review by cag185 — COMMENTED** [review]
Looks good to me, I like the idea of using shared classes, I will want to talk with you about how to actually use that later; will that be a derived class from a common parent class or are we both just sharing one singular class? I also noticed engineFailure flag was spelled "engienFailure" in several places in Block.ts, may want to change before integrating. 

Other than that I like the additions, did you generate the JSON data by hand? Specifically for Track Model, there are some properties missing that I would like to add.

**Comment by chrisg564** [comment]
> Looks good to me, I like the idea of using shared classes, I will want to talk with you about how to actually use that later; will that be a derived class from a common parent class or are we both just sharing one singular class? I also noticed engineFailure flag was spelled "engienFailure" in several places in Block.ts, may want to change before integrating.
> 
> Other than that I like the additions, did you generate the JSON data by hand? Specifically for Track Model, there are some properties missing that I would like to add.

@cag185 Good catch. For derived vs shared class we can discuss tn, but I think shared will allowed for best synchronization between classes (including @cwh35 @kjb135-coe modules).

JSON is converted using a random excel to JSON converter. Feel free to modify as needed. Just note that using specific delimiters (ie using a comma or =) will allow for easy splitting of the strings into arrays. See infrastructure tag on block 5 blue.json

File: src/modules/TrackComponents/Block.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  public brokenRailFailure: boolean;
+  public signalFailure: boolean;
+  public engienFailure: boolean;
+
+  // Fields set by setter functions
+  public switch?: Switch;
+  public occupied: boolean;
+
+  /*
+   * TODO: other classes? beacon  and station
```
Comment (chrisg564) [inline src/modules/TrackComponents/Block.ts:14]: @kjb135-coe @cwh35 Idk your plans for beacons, but this might be a good spot to implement it.

**Review by cwh35 — APPROVED** [review]
looks good

**Pull request merged**